PluginsAndFeatures/azure-toolkit-for-intellij/rider/test-integration/org/cases/markup/functionapp/csharp/isolatedWorker/AzureFunctionAppSyntaxHighlightingTest.kt

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2020-2021 JetBrains s.r.o.
+ *
+ * All rights reserved.
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.cases.markup.functionapp.csharp.isolatedWorker
+
+import com.intellij.lang.annotation.HighlightSeverity
+import com.jetbrains.rdclient.testFramework.waitForDaemon
+import com.jetbrains.rider.test.annotations.TestEnvironment
+import com.jetbrains.rider.test.base.BaseTestWithMarkup
+import com.jetbrains.rider.test.enums.CoreVersion
+import org.testng.annotations.Test
+
+@TestEnvironment(coreVersion = CoreVersion.DOT_NET_5)
+class AzureFunctionAppSyntaxHighlightingTest : BaseTestWithMarkup() {
```

Comment:
We spawn a lot of duplicates in tests in this case. Instead, I would suggest to move all tests into some base class and create two inherited classes for 5.0 and 3.1. You can take a look on example in `RiderTemplatesTestCore` in Rider project.

Reply:
Reworked all tests to use this approach (where applicable).